a3fe/configuration/engine_config.py

```diff
@@ -5,298 +5,243 @@
+    )
+    lambda_val: _Optional[float] = _Field(None, description="Lambda value for current stage")
+
+    ### Alchemical files ###
+    morphfile: _Optional[str] = _Field(None, description="Path to morph file containing alchemical transformation")
+    topfile: _Optional[str] = _Field(None, description="Path to topology file for the system")
+    crdfile: _Optional[str] = _Field(None, description="Path to coordinate file for the system")
+
+    boresch_restraints_dictionary: _Optional[str] = _Field(
+        None, 
+        #alias="boresch restraints dictionary",
+        description="Optional string to hold boresch restraints dictionary content"
+    )    
+    ### Extra options ###
+    extra_options: _Dict[str, str] = _Field(default_factory=dict, description="Extra options to pass to the SOMD engine")    
+
+    @_field_validator('runtime')
+    def validate_runtime(cls, v: float, info: _ValidationInfo) -> float:
+        """Validate that runtime is a multiple of time per cycle using Decimal for precise division"""
+        data = info.data
+        if not ('timestep' in data and 'nmoves' in data):
+            return v
+        
+        time_per_cycle = _Decimal(str(data['timestep'])) * _Decimal(str(data['nmoves'])) / _Decimal('1000000')
+        runtime_decimal = _Decimal(str(v))
+        
+        if runtime_decimal % time_per_cycle != 0:
             raise ValueError(
                 f"Runtime must be a multiple of the time per cycle. "
-                f"Runtime: {self.runtime} ns, Time per cycle: {time_per_cycle} ns."
+                f"Runtime: {v} ns, Time per cycle: {float(time_per_cycle)} ns"
             )
-        
-        # Only try to modify the config file if it exists
-        cfg_file = _os.path.join(self.input_dir, "somd.cfg")
-        if _os.path.exists(cfg_file):
-            # Need to modify the config file to set the correction n_cycles
-            n_cycles = self._calculate_ncycles(self.runtime, time_per_cycle)
-            self._set_n_cycles(n_cycles)
-            print(f"Updated ncycles to {n_cycles} in somd.cfg")
+        return float(v)
 
-    constraint: str = _Field("hbonds", description="Constraint type, must be hbonds or all-bonds")
+    @_model_validator(mode="after")
+    def _check_cutoff_values(self):
+        """
+        Issue warnings if the user supplies certain contradictory or unusual combos.
+        """
+        cutoff_type = self.cutoff_type
+        cutoff_distance = self.cutoff_distance
+        rfd = self.reaction_field_dielectric
+
+        # 1) Only warn if user set reaction_field_dielectric != 78.3
+        if cutoff_type == "cutoffperiodic" and rfd != 78.3:
+            self._logger.warning(
+                "You have cutoff_type=cutoffperiodic but set a reaction_field_dielectric. "
+                f"This value ({rfd}) will be ignored by the engine."
+            )
+
+        # 2) Only warn if user picks e.g. cutoff_distance < 6 or > 18
```

Comment:
Cutoff distance is already enforced in the field to be between 6 and 18. Either remove or warn if not between e.g. 6 and 10. Up to you.